src/proxy/outbound.rs

```diff
@@ -350,14 +350,28 @@ impl OutboundConnection {
 
         let us = us.unwrap();
         // For case upstream server has enabled waypoint
-        if !us.workload.waypoint_addresses.is_empty() {
-            let waypoint_address = us.workload.choose_waypoint_address().unwrap();
+        let waypoint_addr = &us.workload.waypoint;
+        if let Some(waypoint_address) = &waypoint_addr.address {
             // Even in this case, we are picking a single upstream pod and deciding if it has a remote proxy.
             // Typically this is all or nothing, but if not we should probably send to remote proxy if *any* upstream has one.
-            let waypoint_workload = match self.pi.workloads.fetch_workload(&waypoint_address).await
-            {
-                Some(wl) => wl,
-                None => return Err(Error::UnknownWaypoint(waypoint_address, downstream)),
+            let wp_ip_addr = match waypoint_address {
```

Comment:
nit: would be nice to abstract this into something that is just `Workload -> Option<Workload>` (taking user workload -> waypoint`)

Reply:
see b024929468b92d121376987ef3330847ecbc3891